### DISCUSSION TITLE
[Branch 사용 안내]

### DISCUSSION THREAD

**Discussion by JuneParkCode** [42seoul-MiniRT/mlx_engine#4] [Announcements]
- 저희 프로젝트의 브랜치는 Git flow 전략을 따르며, 다음과 같이 설정합니다.

1. main  : 
- 완성된 형태의 배포 가능한 프로그램이 들어있는 브랜치
- main 브랜치에서는 hotfix, develop만 브랜치 분기를 합니다.
2. develop 
- 개발중에 있는 브랜치. PR을 날릴 때 해당 브랜치로 날려주세요.
- 아래에 있는 브랜치를 분기할 때, develop으로부터 브랜치를 분기해주세요.   
3. feature
- 새로운 기능을 만들 때 사용하는 브랜치
- `feature/{기능명}`으로 브랜치를 develop 브랜치로부터 분기해주세요
4. fix
- 기능에서 버그가 발견됐을 시에 `fix/{버그 내용}`으로 브랜치를 분기하여 작업해주세요. 
5. refactor
- 완성된 코드에서 리팩토링을 할 때 사용하는 브랜치입니다. 마찬가지로 develop에서 분기하며, `refactor/{Refactor 내용}` 으로 분기해주세요.

- 각 작업은 fork되어있는 개인 로컬 레포에서 작업 후 PR을 통해서 Merge합니다.

**Comment by JuneParkCode** [comment]
개인 레포에서 작업할 때 반드시 원본 레포로부터 동기화를 한 상태에서 작업해주세요.
https://json.postype.com/post/210431

**Comment by JuneParkCode** [comment]
커밋 시에, [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) 을 참고하여 작성해주시면 더 좋을 것 같습니다.
제 개발환경인 vscode에서는 conventional commit + gitmoji 익스텐션을 활용해서 사용하고 있고, vim + cli 환경에서도 양식 정도만 지켜주시면 깔끔한 커밋 히스토리가 가능할 것 같습니다.
@kimminkyeu

**Reply by kimminkyeu** [reply]
네 확인했습니다. 감사합니다.
